WorkflowSwiftUI/Sources/ObservableState.swift

```diff
@@ -0,0 +1,181 @@
+import Foundation
+import IdentifiedCollections
+
+/// A type that emits notifications to observers when underlying data changes.
+///
+/// Conforming to this protocol signals to other APIs that the value type supports observation.
+/// However, applying the ``ObservableState`` protocol by itself to a type doesn’t add observation
+/// functionality to the type. Instead, always use the ``ObservableState()`` macro when adding
+/// observation support to a type.
+#if !os(visionOS)
+public protocol ObservableState: Perceptible {
+    var _$id: ObservableStateID { get }
+    mutating func _$willModify()
+}
+#else
+public protocol ObservableState: Observable {
+    var _$id: ObservableStateID { get }
+    mutating func _$willModify()
+}
+#endif
+
+/// A unique identifier for a observed value.
+public struct ObservableStateID: Equatable, Hashable, Sendable {
+    @usableFromInline
+    var location: UUID {
+        get { storage.id.location }
+        set {
+            if !isKnownUniquelyReferenced(&storage) {
+                storage = Storage(id: storage.id)
+            }
+            storage.id.location = newValue
+        }
+    }
+
+    private var storage: Storage
+
+    private init(storage: Storage) {
+        self.storage = storage
+    }
+
+    public init() {
+        self.init(storage: Storage(id: .location(UUID())))
```

Comment:
Is the UUID here used for uniquing locations? Could we utilize the #file and #line magic values instead?

Reply:
This is a derived file from TCA. I'm not sure exactly how the location bit works, to be honest.